src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,1008 +50,333 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
-                       hhinfo.addrDefault = addrHH;
-                    
-                    if (isSeasonalAddr(addrHH)) {
-                        Date startDate = getStartOrEndDateOfAddrSeasonal(addrHH, true);
-                        Date endDate = getStartOrEndDateOfAddrSeasonal(addrHH, false);
-                        Date dtToday = System.today();
-                        if (dtToday >= startDate && dtToday < endDate) {
-                            hhinfo.addrSeasonal = addrHH;
-                        }
-                    }
-                }
-            }
-        
         	// finally, if no default found, create an empty default
-            if (hhinfo.addrDefault == null) {
-                Address__c addrEmpty = new Address__c();
-                addrEmpty.Parent_Account__c = accId;
-                hhinfo.addrDefault = addrEmpty;
-            }
-        }
-        
-        // some callers just need the HHInfo for each HH Account.
-        if (isCalcHHInfoOnly)
-            return mapAccIdHHInfo;
-
-        // get all Contacts by HH, that might need updating
-        // note that if we have an organizational account, our query will not pick up any contacts (by design!)
-        map<Id, list<Contact>> mapAccIdListCon = new map<Id, list<Contact>>(); 
-        set<Id> setHHId =  mapAccIdAddr.keySet();
-        list<Contact> listCon = [select Id, is_Address_Override__c, Current_Address__c, AccountId 
-            from Contact where 
-                (Account.RecordTypeId = :UTIL_Describe.getHhAccRecTypeID() or Account.RecordTypeId = :UTIL_Describe.getAdminAccRecTypeID())
-                 and AccountId != null and AccountId in :setHHId];
-        for (Contact con : listCon) {
-            list<Contact> listConHH = mapAccIdListCon.get(con.AccountId);
-            if (listConHH == null) {
-                listConHH = new list<Contact>();
-                mapAccIdListCon.put(con.AccountId, listConHH);
-            }
-            listConHH.add(con);
+            createDefaultIfNoneExists(parentId, parentInfo);
         }
         
         // lists of objects to update
-        list<Account> listAccUpdate = new list<Account>();
-        list<Address__c> listAddrUpdate = new list<Address__c>();
-        list<Contact> listConUpdate = new list<Contact>();
+        List<SObject> listParentUpdate = new List<SObject>();
+        List<Address__c> listAddrUpdate = new List<Address__c>();
         
-        // now go through each HH, and do all the work!
-        for (Id accId : listAccId) {
+        // now go through each parent, and do all the work!
+        for (Id parentId : listParentId) {
             
             // pick the address to use
-            HHInfo hhinfo = mapAccIdHHInfo.get(accId);
-            Address__c addrDefault = hhinfo.addrDefault;
-            Address__c addr = hhinfo.addrSeasonal;
+            ParentInfo ParentInfo = mapParentIdParentInfo.get(parentId);
```

Comment:
Variables should use the `lowerCamelCase` naming convention. Please rename accordingly.